machine/v1beta1/types_alibabaprovider.go

```diff
@@ -0,0 +1,315 @@
+	// +optional
+	Name string `name:"diskName,omitempty"`
+
+	// SnapshotID is the ID of the snapshot used to create data disk N. Valid values of N: 1 to 16.
+	//
+	// When the DataDisk.N.SnapshotID parameter is specified, the DataDisk.N.Size parameter is ignored. The data disk is created based on the size of the specified snapshot.
+	// Use snapshots created after July 15, 2013. Otherwise, an error is returned and your request is rejected.
+	// +optional
+	SnapshotID string `name:"snapshotId,omitempty"`
+
+	// Size of the data disk N. Valid values of N: 1 to 16. Unit: GiB. Valid values:
+	//
+	// Valid values when DataDisk.N.Category is set to cloud_efficiency: 20 to 32768
+	// Valid values when DataDisk.N.Category is set to cloud_ssd: 20 to 32768
+	// Valid values when DataDisk.N.Category is set to cloud_essd: 20 to 32768
+	// Valid values when DataDisk.N.Category is set to cloud: 5 to 2000
+	// The value of this parameter must be greater than or equal to the size of the snapshot specified by the SnapshotID parameter.
+	// +optional
+	Size int `name:"size,omitempty"`
+
+	//Specifies whether to encrypt data disk N.
+	//
+	//Default value: false.
+	// +optional
+	Encrypted bool `name:"encrypted,omitempty"`
+
+	//
+	// PerformanceLevel of the ESSD used as data disk N. The N value must be the same as that in DataDisk.N.Category when DataDisk.N.Category is set to cloud_essd. Default value: PL1. Valid values:
+	//
+	// PL0: A single ESSD can deliver up to 10,000 random read/write IOPS.
+	// PL1: A single ESSD can deliver up to 50,000 random read/write IOPS.
+	// PL2: A single ESSD can deliver up to 100,000 random read/write IOPS.
+	// PL3: A single ESSD can deliver up to 1,000,000 random read/write IOPS.
+	// For more information about ESSD performance levels, see ESSDs.
+	// +optional
+	PerformanceLevel string `name:"performanceLevel,omitempty"`
+
+	// Description of the data disk N. The description must be 2 to 256 characters in length and cannot start with http:// or https://.
+	//
+	// This parameter is empty by default.
+	// +optional
+	Description string `name:"description,omitempty"`
+
+	// Category describes the type of data disk N.
+	// Valid values:
+	// cloud_efficiency: ultra disk
+	// cloud_ssd: standard SSD
+	// cloud_essd: ESSD
+	// cloud: basic disk
+	// For I/O optimized instances, the default value is cloud_efficiency. For non-I/O optimized instances, the default value is cloud.
+	// +optional
+	Category string `name:"category,omitempty"`
+
+	// KMSKeyID is the ID of the Key Management Service (KMS) key to be used by data disk N.
+	// +optional
+	KMSKeyID string `name:"kmsKeyId,omitempty"`
+
+	// Device describes the mount point of data disk N.
+	// +optional
+	Device string `name:"device,omitempty"`
+
+	// DeleteWithInstance specifies whether to release data disk N along with the instance.
+	// Default value: true.
+	// +optional
+	DeleteWithInstance *bool `name:"deleteWithInstance,omitempty"`
```

Comment:
Bool won't pass API review, we will need to change this to an enum, with something like

```
type AlibabaDiskPreservationPolicy string

const (
  DeleteWithInstance AlibabaDiskPreservationPolicy = "DeleteWithInstance"
  PreserveDisk AlibabaDiskPreservationPolicy = "PreserveDisk"
)
```

Reply:
Added.